pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -162,6 +166,93 @@ function subvolume_create(volume, subvol, parent_dir) {
     });
 }
 
+function subvolume_delete(volume, subvol, mount_point_in_parent) {
+    const block = client.blocks[volume.path];
+    const subvols = client.uuids_btrfs_subvols[volume.data.uuid];
+
+    function get_direct_subvol_children(subvol) {
+        function is_direct_parent(sv) {
+            return (sv.pathname.length > subvol.pathname.length &&
+                        sv.pathname.substring(0, subvol.pathname.length) == subvol.pathname &&
+                        sv.pathname[subvol.pathname.length] == "/" &&
+                        sv.pathname.substring(subvol.pathname.length + 1).indexOf("/") == -1);
+        }
+
+        return subvols.filter(is_direct_parent);
+    }
+
+    function get_subvol_children(subvol) {
+        // The deepest nested children must come first
+        const direct_children = get_direct_subvol_children(subvol);
+        return flatten(direct_children.map(get_subvol_children)).concat(direct_children);
+    }
+
+    function remove_configs() {
+        return for_each_async(configs_to_remove, c => block.RemoveConfigurationItem(c, {}));
+    }
+
+    function delete_all_subvols() {
+        return cockpit.spawn(["btrfs", "subvolume", "delete"].concat(paths_to_delete),
+                             { superuser: true, err: "message" });
+    }
+
+    const all_subvols = get_subvol_children(subvol).concat([subvol]);
+    const usage = [];
+    const configs_to_remove = [];
+    const paths_to_delete = [];
+
+    for (const sv of all_subvols) {
+        const [config, mount_point] = get_fstab_config_with_client(client, block, false, sv);
+        const fs_is_mounted = is_mounted(client, block, sv);
+
+        if (fs_is_mounted) {
+            usage.push({
+                level: 0,
+                usage: 'mounted',
+                block,
+                name: sv.pathname,
+                location: mount_point,
+                actions: [_("unmount")],
+                blocking: false,
+            });
+        }
+
+        usage.push({
```

Comment:
I'd say the "unmount" and "delete" entries should be combined into one that has either "unmount, delete" as the action, or just "delete".

Reply:
How it looks now:

![image](https://github.com/cockpit-project/cockpit/assets/67428/ab677a72-f036-4a47-ac3e-3c46c31c9f3f)

Reply:
Yes, that's how it also works for partitions, for example.

![image](https://github.com/cockpit-project/cockpit/assets/3228183/e6a7d709-9abe-489a-a036-7240ea6250f2)